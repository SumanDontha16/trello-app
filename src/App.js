import React, { useState, useEffect } from 'react'
import isEmpty from "lodash/isEmpty";
import axios from "axios";
import Board from "react-trello";

function App() {
  const [cards, setCards] = useState([])

  useEffect(() => {
    async function fetchData() {
      await axios.get(
        'https://jsonplaceholder.typicode.com/users',
      ).then(res => {
        const data = {
          lanes: [
            {
              id: "applicants",
              title: "Applicants",
              style: { width: 280 },
              cards: res.data.map(user => ({
                id: user.id.toString(),
                title: user.name,
                description: user.email
              }))
            },
            {
              id: "interviewed",
              title: "Interviewed",
              style: { width: 280 },
              cards: []
            }
          ]
        };
        setCards(data);
      })
        .catch(error => {
          console.error(error);
        });
    }
    fetchData()
  }, []);

  return (
    <div className="container-fluid">
      <h1 className='mb-3 mt-3' data-testid="title">My Trello App</h1>
      {!isEmpty(cards) ? <Board data={cards} draggable editable canAddLanes addLaneTitle addCardTitle /> : <p>Loading...</p>}
    </div>
  );
}

export default App;
