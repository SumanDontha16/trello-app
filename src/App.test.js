import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import App from "./App";
import renderer from "react-test-renderer";

describe('App render successfully', () => {
  it("Test to match snapshot of component", () => {
    const appTree = renderer
      .create(<App />)
      .toJSON();
    expect(appTree).toMatchSnapshot();
  });

  it("Should renders with correct title", () => {
    render(<App />);
    const headerEl = screen.getByTestId("title");
    expect(headerEl.textContent).toBe("My Trello App");
  });
})