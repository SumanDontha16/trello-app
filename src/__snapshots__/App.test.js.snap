// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render successfully Test to match snapshot of component 1`] = `
<div
  className="container-fluid"
>
  <h1
    className="mb-3 mt-3"
    data-testid="title"
  >
    My Trello App
  </h1>
  <p>
    Loading...
  </p>
</div>
`;
